### PR TITLE
feat: add Duration component

### PR DESCRIPTION
**Pull request by osbornm**
image

### CODE REVIEW

File: src/components/Duration/Duration.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  displaySecondsUntilInMs?: number;
+  /**
+   * The control will display minutes until this number of milliseconds is reached, then hours are used.
+   */
+  displayMinutesUntilInMs?: number;
+  /**
+   * The control will display hours until this number of milliseconds is reached, then days are used.
+   */
+  displayHoursUntilInMs?: number;
```
[the remark this answers is not in the document]
Comment (osbornm) [reply src/components/Duration/Duration.tsx:45]: `displayMinutes` `displayHours` `displayDays`

Call out in description that it used MS and that it is inclusive.

File: src/components/Duration/Duration.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+export const Duration: FC<DurationProps> = ({
+  milliseconds = 0,
+  seconds = 0,
+  minutes = 0,
+  as = 'span',
+  displaySecondsUntilInMs = 60000, // 1 minute
+  displayMinutesUntilInMs = 7200000, // 2 hours
```
Comment (osbornm) [inline src/components/Duration/Duration.tsx:54]: Change to 1 hour

**Comment by nathanyoung** [comment]
[Review Duration component PR](https://trello.com/c/Aszva1Nc/1289-review-duration-component-pr)

**Comment by juanfabrega** [comment]
@osbornm @nathanyoung have we discussed whether this needs to handle internationalization? I know that need continues to grow across the company and I believe all other components either don't deal with hardcoded strings, or provide a way for the user to localize these texts (datepicker for example). I am not fully opposed to merging as is but have some hesitation and would prefer to talk it out. 

Also since it is assumed that the user will be providing basically all styling in the form of `className` + `as`, this functionality is starting to feel more intuitive as a custom hook rather than a component. That would allow users to get these same values but use them inside their already styled markup.

Looking forward to your thoughts on this.

**Comment by osbornm** [comment]
Spoke with @juanfabrega via zoom. Going to add config to allow a user to customize the labels, but we are going to leave as a component for now.

**Review by juanfabrega — APPROVED** [review]
awesome!

**Pull request merged**